frontend/src/Board.tsx

```diff
@@ -1,37 +1,47 @@
+import { CardState, InputValue } from "./App";
 import Card from "./Card";
 
-const cardPositions = {
-    0: {'top': '20px', 'left': '20px'},
-    1: {'top': '20px', 'left': '185px'},
-    2: {'top': '20px', 'left': '350px'},
-    3: {'top': '20px', 'left': '520px'},
-    4: {'top': '220px', 'left': '20px'},
-    5: {'top': '220px', 'left': '185px'},
-    6: {'top': '220px', 'left': '350px'},
-    7: {'top': '220px', 'left': '520px'},
-    8: {'top': '420px', 'left': '20px'},
-    9: {'top': '420px', 'left': '185px'},
-    10: {'top': '420px', 'left': '350px'},
-    11: {'top': '420px', 'left': '520px'}
+const cardPositions = new Map<number, { top: string, left: string}>()
+  .set(0, {'top': '20px', 'left': '20px'})
+  .set(1, {'top': '20px', 'left': '185px'})
+  .set(2, {'top': '20px', 'left': '350px'})
+  .set(3, {'top': '20px', 'left': '520px'})
+  .set(4, {'top': '220px', 'left': '20px'})
+  .set(5, {'top': '220px', 'left': '185px'})
+  .set(6, {'top': '220px', 'left': '350px'})
+  .set(7, {'top': '220px', 'left': '520px'})
+  .set(8, {'top': '420px', 'left': '20px'})
+  .set(9, {'top': '420px', 'left': '185px'})
+  .set(10, {'top': '420px', 'left': '350px'})
+  .set(11, {'top': '420px', 'left': '520px'})
```

Comment:
I've changed the use of a POJO (plain old javascript object) to a Map here for two reasons. As I mentioned above, typescript gets grumpy when we access objects dynamically with `[]` notation, even though JavaScript is totally ok with it.

Because `cardPositions` is a POJO and not a class, I opted to use a Map. This makes TypeScript happy as it can infer all of the key and value types correctly. 

The other reason I opted for a Map is because that's how it's being used. Not only do we get the added benefit of adding a little bit of self-documenting code to the codebase, but we also get performance improvements as V8 (the runtime in Node & most browsers that runs our JavaScript) knows how to optimise Maps as they have a very narrow use-case as compared to POJOs.